File: Codes/Hash/Non-Cryptographic/DJBHash/DJBHash.java
/*
	Hash function by Daniel J. Bernstein
    Archive of Reversing.ID
    Non-Cryptographic Hash
*/

// Fungsi ini merupakan method dari sebuah class
public long DJBHash(String key)
{
    long state = 5381;

    for(int i = 0; i < key.length(); i++)
    {
        state = ((state << 5) + state) + key.charAt(i);
    }

    return state;
}

File: Codes/Hash/Non-Cryptographic/DJBHash/DJBHash.rb
# Hash function by Daniel J. Bernstein
# Archive of Reversing.ID
# Non-Cryptographic Hash

def djbhash( key, len=key.length )
    state = 5381
    
    len.times{ |i|
        state = ((state << 5) + state) + key[i]
    }
    return state
end